File: code/713/solution.go
package main /*{{{*/

import (
	"fmt"
	"math"
)

func L(n int64) int64 {
	m := int64(math.Sqrt(float64(n - 1)))
	ret := m*m*(m+1)*(m+1)/4 - m*m*n

	for i := m; i >= 1; i-- {
		ret += (n - 1) / i * (2*n - i*(1+(n-1)/i))
	}
	return ret
}

func main() {
	fmt.Println(L(1000))
	fmt.Println(L(1e7))
}

File: code/713/solution.cpp
#include <iostream>/*{{{*/
#include <math.h>
using namespace std; /*}}}*/
long long L(long long n) {
    long long m   = sqrt(n - 1);
    long long ret = m * m * (1 + m) * (1 + m) / 4 - m * m * n;
    for (int i = 1; i <= m; i++) {
        ret += (n - 1) / i * (2 * n - i * (1 + (n - 1) / i));
    }
    return ret;
}
int main() {
    // Sum_{k=1..n-1} Sum_{i=1..n-1} floor(k/i)
    cout << L(1000) << endl;
    cout << L(1e7) << endl;
    return 0;
}
